pkg/operator/ceph/csi/config.go

```diff
@@ -0,0 +1,174 @@
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			err = c.Create(ctx, csiOpClientProfile)
+			if err != nil {
+				return errors.Wrapf(err, "failed to create ceph-csi clientProfile cr for RBD %q", csiOpClientProfile.Name)
+			}
+			logger.Infof("successfully created ceph-csi clientProfile CR for RBD %q", csiOpClientProfile.Name)
+			return nil
+		}
+		return err
+	}
+
+	err = c.Update(ctx, csiOpClientProfile)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create ceph-csi clientProfile cr for RBD %q", csiOpClientProfile.Name)
+	}
+	logger.Infof("successfully updated ceph-csi clientProfile CR for RBD %q", csiOpClientProfile.Name)
+
+	return nil
+}
+
+func CreateUpdateClientProfileSubVolumeGroup(ctx context.Context, c client.Client, clusterInfo *cephclient.ClusterInfo, cephFilesystemNamespacedName types.NamespacedName, clusterID, clusterName string) error {
+
+	logger.Info("Creating ceph-csi clientProfile CR for subvolume group")
+
+	csiOpClientProfile := generateProfileSubVolumeGroupSpec(clusterInfo, cephFilesystemNamespacedName, clusterID, clusterName)
+
+	err := clusterInfo.OwnerInfo.SetOwnerReference(csiOpClientProfile)
+	if err != nil {
+		return errors.Wrapf(err, "failed to set owner reference for clientProfile CR %q for subVolGrp", csiOpClientProfile.Name)
+	}
+
+	err = c.Get(ctx, types.NamespacedName{Name: csiOpClientProfile.Name, Namespace: csiOpClientProfile.Namespace}, csiOpClientProfile)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			err = c.Create(ctx, csiOpClientProfile)
+			if err != nil {
+				return errors.Wrapf(err, "failed to create ceph-csi clientProfile cr for subVolGrp %q", csiOpClientProfile.Name)
+			}
+			logger.Infof("successfully created ceph-csi clientProfile CR for subVolGrp %q", csiOpClientProfile.Name)
+			return nil
+		}
+		return err
+	}
+
+	err = c.Update(ctx, csiOpClientProfile)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create ceph-csi clientProfile cr for subVolGrp %q", csiOpClientProfile.Name)
+	}
+	logger.Infof("successfully updated ceph-csi clientProfile CR for subVolGrp %q", csiOpClientProfile.Name)
+
+	return nil
+}
+
+func generateProfileSubVolumeGroupSpec(clusterInfo *cephclient.ClusterInfo, cephFilesystemNamespacedName types.NamespacedName, clusterID, clusterName string) *csiopv1a1.ClientProfile {
+	csiOpClientProfile := &csiopv1a1.ClientProfile{}
+	csiOpClientProfile.Name = clusterID
+	csiOpClientProfile.Namespace = cephFilesystemNamespacedName.Namespace
+	csiOpClientProfile.Spec = csiopv1a1.ClientProfileSpec{
+		CephConnectionRef: v1.LocalObjectReference{
+			Name: clusterName,
+		},
+		CephFs: &csiopv1a1.CephFsConfigSpec{
+			SubVolumeGroup: cephFilesystemNamespacedName.Name,
+		},
+	}
+
+	kernelMountKeyVal := strings.Split(clusterInfo.CSIDriverSpec.CephFS.KernelMountOptions, "=")
+	fuseMountKeyVal := strings.Split(clusterInfo.CSIDriverSpec.CephFS.FuseMountOptions, "=")
+
+	if len(kernelMountKeyVal) == 2 {
```

Comment:
@Madhu-1 There could be multiple kernel and fuse mount options, separated by commas, right? This assumes there is only a single option. A follow-up PR is fine for this though.